File: frontend/src/types/photo.ts
export interface Photo {
  id: number
  userId: number
  albumId: number
  albumName: string
  title: string
  description: string
  filePath: string
  thumbnailPath: string
  fileSize: number
  fileType: string
  width: number
  height: number
  shootTime: string
  cameraModel: string
  gpsLatitude: number
  gpsLongitude: number
  isFavorite: number
  viewCount: number
  createdAt: string
  tags: Tag[]
}

export interface PhotoQuery {
  pageNum: number
  pageSize: number
  albumId?: number
  keyword?: string
  startTime?: string
  endTime?: string
  tagId?: number
  isFavorite?: number
}

export interface PhotoUpdate {
  id: number
  title: string
  description?: string
  albumId?: number
  tagIds?: number[]
}

File: backend/src/main/java/com/photo/entity/Photo.java
package com.photo.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("t_photo")
public class Photo implements Serializable {
    @TableId(type = IdType.AUTO)
    private Long id;

    private Long userId;

    private Long albumId;

    private String title;

    private String description;

    private String filePath;

    private String thumbnailPath;

    private Long fileSize;

    private String fileType;

    private Integer width;

    private Integer height;

    private LocalDateTime shootTime;

    private String cameraModel;

    private String location;

    private BigDecimal gpsLatitude;

    private BigDecimal gpsLongitude;

    private Integer isDeleted;

    private Integer isFavorite;

    private Integer isPrivate;

    private String privatePassword;

    private Integer viewCount;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}
